service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -260,10 +262,111 @@ public void createCloudContext(
   }
 
   /**
-   * Delete a cloud context for the workspace validating the flight id. This is for use by the cloud
-   * context create flight. We compare the incoming flight id with the creating_flight of the cloud
-   * context. We will only delete if they match. That makes sure that undoing a create does not
-   * delete a valid cloud context.
+   * Create cloud context - this is used as part of CreateGcpContextFlightV2 to insert the context
+   * row at the start of the create context operation.
+   */
+  @WriteTransaction
+  public void createCloudContext(UUID workspaceId, CloudPlatform cloudPlatform, String flightId) {
+    final String platform = cloudPlatform.toSql();
+    final String sql =
+        "INSERT INTO cloud_context (workspace_id, cloud_platform, creating_flight)"
+            + " VALUES (:workspace_id, :cloud_platform, :creating_flight)";
+    MapSqlParameterSource params =
+        new MapSqlParameterSource()
+            .addValue("workspace_id", workspaceId.toString())
+            .addValue("cloud_platform", platform)
+            .addValue("creating_flight", flightId);
+    try {
+      jdbcTemplate.update(sql, params);
+      logger.info("Inserted record for {} cloud context for workspace {}", platform, workspaceId);
+    } catch (DuplicateKeyException e) {
+      throw new DuplicateCloudContextException(
+          String.format(
+              "Workspace with id %s already has context for %s cloud type", workspaceId, platform),
+          e);
+    }
+  }
+
+  /**
+   * Second part of the create cloud context - write the context. This transaction is run from a
+   * flight step, so we want it to be idempotent. The algorithm is this:
+   *
+   * <ol>
+   *   <li>try the update, searching explicitly for our created flight
+   *   <li>if nothing gets updated, then maybe this is a step restart. We query the row to make sure
+   *       that it exists and has a non-null context.
+   * </ol>
+   *
+   * Since the typical case will be a successful update, we won't do the get frequently.
+   *
+   * @param workspaceId workspaceId part of PK for lookup
+   * @param cloudPlatform platform part of PK for lookup
+   * @param context serialized cloud context
+   * @param flightId flight id
+   */
+  @WriteTransaction
+  public void updateCloudContext(
```

Comment:
It's a little confusing to me that we have two `updateCloudContext` methods, where one is really finalizing creation and the other is actually used to update from V1 to V2. Maybe rename this one, or maybe just add javadoc on the one used to upgrade versions?